modules/ximgproc/src/sparse_match_interpolators.cpp

```diff
@@ -877,5 +888,1148 @@ bool operator<(const SparseMatch& lhs,const SparseMatch& rhs)
+        }
+
+        m_index = new float[size];
+        m_weight = new float[size];
+        m_size = size;
+        m_validSize = 0;
+
+        return 0;
+    }
+    int Push(float index, float weight)
+    {
+        if (m_validSize >= m_size)
+        {
+            CV_Error(CV_StsOutOfRange, " m_validSize >= m_size this problem can be resolved my decreasig k parameter");
+        };
+        m_index[m_validSize] = index;
+        m_weight[m_validSize] = weight;
+        m_validSize++;
+
+        int i = m_validSize - 1;
+        while (prior(m_weight[i], m_weight[(i - 1) / 2])) {
+            swap(m_weight[i], m_weight[(i - 1) / 2]);
+            swap(m_index[i], m_index[(i - 1) / 2]);
+            i = (i - 1) / 2; // jump up to the parent
+        }
+
+        return i;
+    }
+
+    float Pop(float* weight = NULL)
+    {
+        if (m_validSize == 0) {
+            return -1;
+        }
+
+        if (weight) {
+            *weight = m_weight[0];
+        }
+        float outIdx = m_index[0];
+
+        // use the last item to overwrite the first
+        m_index[0] = m_index[m_validSize - 1];
+        m_weight[0] = m_weight[m_validSize - 1];
+        m_validSize--;
+
+        // adjust the heap from top to bottom
+        float rawIdx = m_index[0];
+        float rawWt = m_weight[0];
+        int candiPos = 0; // the root
+        int i = 1; // left child of the root
+        while (i < m_validSize) {
+            // test right child
+            if (i + 1 < m_validSize && prior(m_weight[i + 1], m_weight[i])) {
+                i++;
+            }
+            if (prior(rawWt, m_weight[i])) {
+                break;
+            }
+            m_index[candiPos] = m_index[i];
+            m_weight[candiPos] = m_weight[i];
+            candiPos = i;
+
+            i = (i + 1) * 2 - 1; // left child
+        }
+        m_index[candiPos] = rawIdx;
+        m_weight[candiPos] = rawWt;
+
+        return outIdx;
+    }
+    void Clear()
+    {
+        m_validSize = 0;
+    }
+    int Size()
+    {
+        return m_validSize;
+    }
+
+private:
+    inline void swap(float & a, float & b)
+    {
+        float c = a;
+        a = b;
+        b = c;
+    }
+    inline bool prior(float w1, float w2)
+    {
+            return w1 < w2;
+    }
+
+    float* m_index;
+    float* m_weight;
+    int m_size;
+    int m_validSize;
+};
+
+void RICInterpolatorImpl::interpolate(InputArray from_image, InputArray from_points, InputArray to_image, InputArray to_points, OutputArray dense_flow)
+{
+    CV_Assert(!from_image.empty() && (from_image.depth() == CV_8U) && (from_image.channels() == 3 || from_image.channels() == 1));
+    CV_Assert( use_variational_refinement == false || (!to_image.empty() && (to_image.depth() == CV_8U) && (to_image.channels() == 3 || to_image.channels() == 1)));
+    CV_Assert(!from_points.empty() && from_points.isVector() &&
+        !to_points.empty() && to_points.isVector() &&
+        from_points.sameSize(to_points));
```

Comment:
Avoid using of complex assertion statements - they emits over-complicated unclear messages to Users.

```
CV_Assert(!from_points.empty() && from_points.isVector());
CV_Assert(!to_points.empty() && to_points.isVector());
CV_Assert(from_points.sameSize(to_points));
```

Their messages helping to localize parameters mistakes much better.

Reply:
done